Plugins publish 'workspace:*' dep — uninstallable from registry

## Problem

The plugin packages ship with an unresolved `workspace:*` protocol in their **published** `dependencies`, so they can't be installed outside the monorepo.

```
$ pnpm add -D @karnak19/pbkit-tanstack
 ERR_PNPM_WORKSPACE_PKG_NOT_FOUND  In : "@karnak19/pbkit@workspace:*" is in the
 dependencies but no package named "@karnak19/pbkit" is present in the workspace
```

Confirmed on the latest published versions:

| package | version | dependencies |
|---|---|---|
| `@karnak19/pbkit-tanstack` | 0.2.2 | `{ "@karnak19/pbkit": "workspace:*" }` |
| `@karnak19/pbkit-zod` | 0.0.4 | `{ "@karnak19/pbkit": "workspace:*" }` |
| `@karnak19/pbkit-realtime` | 0.1.1 | `{ "@karnak19/pbkit": "workspace:*" }` |

`workspace:*` is meant to be rewritten to a real semver range at publish time. It wasn't, so npm/pnpm consumers fail to resolve it.

## Impact

All three plugins are uninstallable from the registry. For a consuming app, that blocks adopting `pbkit-tanstack` (React Query layer), `pbkit-zod`, and `pbkit-realtime` entirely — i.e. everything beyond core type generation.

## Fix

`workspace:*` should resolve to a real range on publish. Options:
- If using **changesets** / `pnpm publish`: ensure publishing goes through the tool that rewrites `workspace:` protocols (pnpm does this automatically on `pnpm publish`; publishing via raw `npm publish` skips the rewrite). 
- Or set the dependency explicitly, e.g. `"@karnak19/pbkit": "^0.4.0"`.
- Consider moving `@karnak19/pbkit` to **`peerDependencies`** on the plugins (they're meant to run alongside a specific core version anyway), with it also in devDeps for local builds.

## Verify

After republishing: `pnpm add -D @karnak19/pbkit-tanstack` in a fresh project (no workspace) should resolve and install cleanly.

--- a/.changeset/config.json
+++ b/.changeset/config.json
@@ -7,5 +7,8 @@
   "access": "public",
   "baseBranch": "main",
   "updateInternalDependencies": "patch",
-  "ignore": []
+  "ignore": [],
+  "___experimentalUnsafeOptions_WILL_CHANGE_IN_PATCH": {
+    "onlyUpdatePeerDependentsWhenOutOfRange": true
+  }
 }

--- a/.changeset/plugin-peer-deps.md
+++ b/.changeset/plugin-peer-deps.md
@@ -1,0 +1,7 @@
+---
+"@karnak19/pbkit-tanstack": patch
+"@karnak19/pbkit-zod": patch
+"@karnak19/pbkit-realtime": patch
+---
+
+Fix uninstallable plugins by moving `@karnak19/pbkit` from `dependencies` (`workspace:*`) to `peerDependencies` (`^0.4.0`). The unresolved `workspace:*` protocol was being published verbatim, so the plugins couldn't be installed outside the monorepo. The core package is now declared as a peer dependency — consumers already install it as the host CLI — which also guarantees the plugin shares the host's single pbkit instance.

--- a/packages/pbkit-realtime/package.json
+++ b/packages/pbkit-realtime/package.json
@@ -37,12 +37,11 @@
   },
   "license": "MIT",
   "peerDependencies": {
+    "@karnak19/pbkit": "^0.4.0",
     "pocketbase": ">=0.21.0"
   },
-  "dependencies": {
-    "@karnak19/pbkit": "workspace:*"
-  },
   "devDependencies": {
+    "@karnak19/pbkit": "workspace:*",
     "@types/bun": "^1.3.13",
     "typescript": "^6.0.0"
   }

--- a/packages/pbkit-tanstack/package.json
+++ b/packages/pbkit-tanstack/package.json
@@ -37,12 +37,11 @@
   },
   "license": "MIT",
   "peerDependencies": {
+    "@karnak19/pbkit": "^0.4.0",
     "@tanstack/query-core": "^5.0.0"
   },
-  "dependencies": {
-    "@karnak19/pbkit": "workspace:*"
-  },
   "devDependencies": {
+    "@karnak19/pbkit": "workspace:*",
     "@types/bun": "^1.3.13",
     "typescript": "^6.0.0"
   }

--- a/packages/pbkit-zod/package.json
+++ b/packages/pbkit-zod/package.json
@@ -37,12 +37,11 @@
   },
   "license": "MIT",
   "peerDependencies": {
+    "@karnak19/pbkit": "^0.4.0",
     "zod": "^3.0.0"
   },
-  "dependencies": {
-    "@karnak19/pbkit": "workspace:*"
-  },
   "devDependencies": {
+    "@karnak19/pbkit": "workspace:*",
     "@types/bun": "^1.3.13",
     "typescript": "^6.0.0",
     "zod": "^3.25.0"

